python/google/protobuf/message_factory.py

```diff
@@ -122,6 +115,32 @@ def GetMessages(self, files):
         extended_class.RegisterExtension(extension)
     return result
 
+  def ClearCache(self):
```

Comment:
Please do not add new APIs for fixing PR. 

If you want to add new APIs, we should have a API review (as a public library, we are very careful to add new APIs). New API should be supported in both pure python and cpp extension. Also need to have test coverage for it.

Reply:
Done.